migrations/V192__Add_Common_Milestone_Solution_Link_Library.sql

```diff
@@ -0,0 +1,165 @@
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('HELPDESK_SUPPORT', '{CBOSC}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('IDDOC_SUPPORT', '{INNOVATION, ACO_OS}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('ESTABLISH_BENCH', '{IDR, CCW}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('PROCESS_PART_APPEALS', '{LV}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('ACQUIRE_AN_EVAL_CONT', '{RMADA}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('DATA_TO_MONITOR', '{CDX, CCW, HDR, IDR, POST_PORTAL}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('DATA_TO_SUPPORT_EVAL', '{CDX, CCW, HDR, POST_PORTAL}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('CLAIMS_BASED_MEASURES', '{IDR, CCW}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('QUALITY_PERFORMANCE_SCORES', '{EXISTING_CMS_DATA_AND_PROCESS, HDR}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('SEND_REPDATA_TO_PART', '{INNOVATION, BCDA, CDX, CMS_BOX, EFT, EDFR, RMADA, POST_PORTAL}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('UTILIZE_QUALITY_MEASURES_DEVELOPMENT_CONTRACTOR', '{QV, MIDS}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('ACQUIRE_A_LEARN_CONT', '{RMADA}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('PART_TO_PART_COLLAB', '{CONNECT}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('EDUCATE_BENEF', '{LDG}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('IT_PLATFORM_FOR_LEARNING', '{CONNECT}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('ADJUST_FFS_CLAIMS', '{SHARED_SYSTEMS, HIGLAS}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('MANAGE_FFS_EXCL_PAYMENTS', '{SHARED_SYSTEMS}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('MAKE_NON_CLAIMS_BASED_PAYMENTS', '{APPS, HIGLAS, IPC, MDM_NCBP}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('COMPUTE_SHARED_SAVINGS_PAYMENT', '{RMADA}')
+    UNION
+    SELECT
+        "key",
+        "value"
+    FROM link_key_and_array('RECOVER_PAYMENTS', '{APPS, IPC, SHARED_SYSTEMS}')
+)
+
+-- Insert into common milestone link from the union select that combines the keys
+
+INSERT INTO mto_common_milestone_solution_link (mto_common_milestone_key, mto_common_solution_key)
+SELECT
+    cast( key AS MTO_COMMON_MILESTONE_KEY) AS milestone,
+    cast( value AS MTO_COMMON_SOLUTION_KEY) AS solution
```

Comment:
You know, I originally did that approach and decided against it. I did something very similar to that when we were making the original it solution implementation, but that required me to create another function now instead of reusing it. 

This seemed to me a good solution that allows for greater flexibility. I figured in the end we will always be casting sometime,  but if we do it in the function, it's not very flexible. Previously I did an insert in each of these functions, but doing a table level operation seemed to be a more flexible solution to me.